x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/indexlifecycle/SetSingleNodeAllocateStepTests.java

```diff
@@ -340,17 +360,24 @@ private void assertNodeSelected(IndexMetaData indexMetaData, Index index, Set<St
 
         Mockito.when(client.admin()).thenReturn(adminClient);
         Mockito.when(adminClient.indices()).thenReturn(indicesClient);
+
+        final UpdateSettingsResponse response;
+        try (XContentParser parser = XContentType.JSON.xContent()
+                .createParser(NamedXContentRegistry.EMPTY, THROW_UNSUPPORTED_OPERATION, "{\"acknowledged\": true}")) {
+            response = UpdateSettingsResponse.fromXContent(parser);
```

Comment:
😄

Reply:
I ended up using the `UpdateSettingsAction.INSTANCE.newResponse()` then `response.readFrom(StreamInput.wrap(new byte[] { ... }))` route for this workaround. But I like this too! more self-describing

Reply:
Can we not make the Response constructor public here? Having the object parsed from a JSON string seems fragile and ugly to me.